web_client/dialogs/editStyleGroups.js

```diff
@@ -0,0 +1,162 @@
+import tinycolor from 'tinycolor2';
+
+import View from 'girder/views/View';
+
+import StyleModel from '../models/StyleModel';
+import editStyleGroups from '../templates/dialogs/editStyleGroups.pug';
+import 'girder/utilities/jquery/girderModal';
+import '../stylesheets/dialogs/editStyleGroups.styl';
+
+/**
+ * Create a modal dialog with fields to edit and create annotation
+ * style groups.
+ */
+const EditStyleGroups = View.extend({
+    events: {
+        'click .h-create-new-style': '_createNewStyle',
+        'click .h-save-new-style': '_saveNewStyle',
+        'change .h-style-def': '_updateStyle',
+        'changeColor .h-colorpicker': '_updateStyle',
+        'change select': '_setStyle'
+    },
+
+    render() {
+        this.$('.h-colorpicker').colorpicker('destroy');
+        this.$el.html(
+            editStyleGroups({
+                collection: this.collection,
+                model: this.model,
+                newStyle: this._newStyle
+            })
+        );
+        this.$('.h-colorpicker').colorpicker();
+        return this;
+    },
+
+    _setStyle(evt) {
+        evt.preventDefault();
+        this.model.set(
+            this.collection.get(this.$('.h-style-selector').val()).toJSON()
+        );
+        this.render();
+    },
+
+    _updateStyle(evt) {
+        evt.preventDefault();
+
+        const data = {};
+        const label = this.$('#h-element-label').val();
+        let validation = '';
+
+        data.id = this.$('.h-style-selector :selected').val() || this.$('.h-new-style-name').val();
+        if (!data.id) {
+            validation += 'A style name is required';
+            this.$('.h-new-style-name').parent().addClass('has-error');
+        }
+
+        if (label) {
```

Comment:
It would be nice to be able to unset the label (though that could wait for another PR).  Otherwise, once a label is added, you can never go back to the no-label mode.